boofuzz/sessions.py

```diff
@@ -140,10 +140,10 @@ def send(self, data):
         Returns:
             None
         """
-        if self._fuzz_data_logger is not None:
-            self._fuzz_data_logger.log_send(data)
+        num_sent, data_sent = self._target_connection.send(data=data)
 
-        num_sent = self._target_connection.send(data=data)
+        if self._fuzz_data_logger is not None:
+            self._fuzz_data_logger.log_send(data_sent)
```

Comment:
We could get a similar effect with something like:
```python
            self._fuzz_data_logger.log_send(data[:num_sent])
```
If that works, we could also skip the changes to the way the send function works.

Reply:
Great idea, this works perfectly fine!